### DISCUSSION TITLE
设置了XMAKE_MAIN_REPO为gitee下载还是github

### DISCUSSION THREAD

**Discussion by MichaelWU0726** [xmake-io/xmake#2085] [Q&A]
设置了XMAKE_MAIN_REPO为gitee

```bash
$ echo $XMAKE_MAIN_REPO
https://gitee.com/tboox/xmake-repo.git
```
下载包时还是走的github
```bash
error: --2022-02-23 12:15:11--  https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
Resolving github.com (github.com)... failed: Temporary failure in name resolution.
wget: unable to resolve host address 'github.com'
```

另外文档排版有误
<img width="1367" alt="屏幕截图 2022-02-23 200721" src="https://user-images.githubusercontent.com/49356156/155323973-2241709c-bca7-4c9d-a3fa-90b482bddec1.png">

**Comment by waruqi** [comment]
看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题

只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip

这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的

只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86

**Comment by MichaelWU0726** [comment]
> 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> 
> 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> 
> 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> 
> 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86

并没有下载openssl啊
```bash
please input: y (y/n/m)
y
  => download https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip .. failed

we can also download these packages manually:
  - https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
to the local search directories: 
  - OpenSSL_1_1_1k.zip
and we can run `xmake g --pkg_searchdirs=/xxx` to set the search directories.
```

**Comment by waruqi** [comment]
>   => download https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip .. failed

这不是在下载么。

**Comment by MichaelWU0726** [comment]
> > => download https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip .. failed
> 
> 这不是在下载么。

首先它是从github下载而不是gitee下载，其次它也是失败啊

**Comment by MichaelWU0726** [comment]
> 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> 
> 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> 
> 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> 
> 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86

意思是MAIN-REPO变量只是设置xmake-repo的包索引仓？而包的实际下载url还是根据包本身的配置信息？

**Comment by waruqi** [comment]
> > 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> > 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> > 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> > 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86
> 
> 意思是MAIN-REPO变量只是设置xmake-repo的包索引仓？而包的实际下载url还是根据包本身的配置信息？

那当然了，xmake 不提供云存储，不可能所有包都去 github 上存一份，包自身源码在哪 依赖包本身的官方源，有可能他们放在 github，也有可能放在 sourceforge gitlab ftp 等各种地方。。怎么可能统一切

**Comment by waruqi** [comment]
> > > => download https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip .. failed
> > 
> > 
> > 这不是在下载么。
> 
> 首先它是从github下载而不是gitee下载，其次它也是失败啊

这跟 XMAKE_MAIN_REPO 没有任何关系，包源自身下载，走包自身 下载源，改不了。。你只能开 mirror 自己配置切换，上面文档有描述

可以看下 openssl 的包源配置

https://github.com/xmake-io/xmake-repo/blob/b2859f319cbfc1a83166de7286186f1db5264a87/packages/o/openssl/xmake.lua#L6

**Comment by MichaelWU0726** [comment]
> > > 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> > > 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> > > 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> > > 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86
> > 
> > 
> > 意思是MAIN-REPO变量只是设置xmake-repo的包索引仓？而包的实际下载url还是根据包本身的配置信息？
> 
> 那当然了，xmake 不提供云存储，不可能所有包都去 github 上存一份，包自身源码在哪 依赖包本身的官方源，有可能他们放在 github，也有可能放在 sourceforge gitlab ftp 等各种地方。。怎么可能统一切

我刚刚清除了.xmake文件夹重新xmake，显示这个：
```bash
checking for platform ... linux
checking for architecture ... x86_64
updating repositories .. error: Cloning into '/home/wq/.xmake/repositories/build-artifacts'...
fatal: unable to access 'https://github.com/xmake-mirror/build-artifacts.git/': Could not resolve host: github.com
```
这个是不是意味着MAIN-REPO没起作用？

**Comment by waruqi** [comment]
> > > > 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> > > > 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> > > > 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> > > > 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86
> > > 
> > > 
> > > 意思是MAIN-REPO变量只是设置xmake-repo的包索引仓？而包的实际下载url还是根据包本身的配置信息？
> > 
> > 
> > 那当然了，xmake 不提供云存储，不可能所有包都去 github 上存一份，包自身源码在哪 依赖包本身的官方源，有可能他们放在 github，也有可能放在 sourceforge gitlab ftp 等各种地方。。怎么可能统一切
> 
> 我刚刚清除了.xmake文件夹重新xmake，显示这个：
> 
> ```shell
> checking for platform ... linux
> checking for architecture ... x86_64
> updating repositories .. error: Cloning into '/home/wq/.xmake/repositories/build-artifacts'...
> fatal: unable to access 'https://github.com/xmake-mirror/build-artifacts.git/': Could not resolve host: github.com
> ```
> 
> 这个是不是意味着MAIN-REPO没起作用？

仔细看文档，二进制包仓库，不是这个 env 配置，https://xmake.io/#/zh-cn/guide/configuration?id=xmake_binary_repo

**Comment by MichaelWU0726** [comment]
> > > > > 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> > > > > 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> > > > > 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> > > > > 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86
> > > > 
> > > > 
> > > > 意思是MAIN-REPO变量只是设置xmake-repo的包索引仓？而包的实际下载url还是根据包本身的配置信息？
> > > 
> > > 
> > > 那当然了，xmake 不提供云存储，不可能所有包都去 github 上存一份，包自身源码在哪 依赖包本身的官方源，有可能他们放在 github，也有可能放在 sourceforge gitlab ftp 等各种地方。。怎么可能统一切
> > 
> > 
> > 我刚刚清除了.xmake文件夹重新xmake，显示这个：
> > ```shell
> > checking for platform ... linux
> > checking for architecture ... x86_64
> > updating repositories .. error: Cloning into '/home/wq/.xmake/repositories/build-artifacts'...
> > fatal: unable to access 'https://github.com/xmake-mirror/build-artifacts.git/': Could not resolve host: github.com
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 这个是不是意味着MAIN-REPO没起作用？
> 
> 仔细看文档，二进制包仓库，不是这个 env 配置，https://xmake.io/#/zh-cn/guide/configuration?id=xmake_binary_repo

二进制包仓库和主仓的env是同一个，同一个变量不能同时两个值吧

**Comment by waruqi** [comment]
> > > > > > 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> > > > > > 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> > > > > > 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> > > > > > 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86
> > > > > 
> > > > > 
> > > > > 意思是MAIN-REPO变量只是设置xmake-repo的包索引仓？而包的实际下载url还是根据包本身的配置信息？
> > > > 
> > > > 
> > > > 那当然了，xmake 不提供云存储，不可能所有包都去 github 上存一份，包自身源码在哪 依赖包本身的官方源，有可能他们放在 github，也有可能放在 sourceforge gitlab ftp 等各种地方。。怎么可能统一切
> > > 
> > > 
> > > 我刚刚清除了.xmake文件夹重新xmake，显示这个：
> > > ```shell
> > > checking for platform ... linux
> > > checking for architecture ... x86_64
> > > updating repositories .. error: Cloning into '/home/wq/.xmake/repositories/build-artifacts'...
> > > fatal: unable to access 'https://github.com/xmake-mirror/build-artifacts.git/': Could not resolve host: github.com
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 这个是不是意味着MAIN-REPO没起作用？
> > 
> > 
> > 仔细看文档，二进制包仓库，不是这个 env 配置，https://xmake.io/#/zh-cn/guide/configuration?id=xmake_binary_repo
> 
> 二进制包仓库和主仓的env是同一个，同一个变量不能同时两个值吧

不是同一个 env ，XMAKE_MAIN_REPO and XMAKE_BINARY_REPO

**Comment by MichaelWU0726** [comment]
> > > > > > > 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> > > > > > > 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> > > > > > > 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> > > > > > > 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86
> > > > > > 
> > > > > > 
> > > > > > 意思是MAIN-REPO变量只是设置xmake-repo的包索引仓？而包的实际下载url还是根据包本身的配置信息？
> > > > > 
> > > > > 
> > > > > 那当然了，xmake 不提供云存储，不可能所有包都去 github 上存一份，包自身源码在哪 依赖包本身的官方源，有可能他们放在 github，也有可能放在 sourceforge gitlab ftp 等各种地方。。怎么可能统一切
> > > > 
> > > > 
> > > > 我刚刚清除了.xmake文件夹重新xmake，显示这个：
> > > > ```shell
> > > > checking for platform ... linux
> > > > checking for architecture ... x86_64
> > > > updating repositories .. error: Cloning into '/home/wq/.xmake/repositories/build-artifacts'...
> > > > fatal: unable to access 'https://github.com/xmake-mirror/build-artifacts.git/': Could not resolve host: github.com
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > 这个是不是意味着MAIN-REPO没起作用？
> > > 
> > > 
> > > 仔细看文档，二进制包仓库，不是这个 env 配置，https://xmake.io/#/zh-cn/guide/configuration?id=xmake_binary_repo
> > 
> > 
> > 二进制包仓库和主仓的env是同一个，同一个变量不能同时两个值吧
> 
> 不是同一个 env ，XMAKE_MAIN_REPO and XMAKE_BINARY_REPO

那就是你文档的代码写错了

**Comment by waruqi** [comment]
> > > > > > > > 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> > > > > > > > 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> > > > > > > > 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> > > > > > > > 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86
> > > > > > > 
> > > > > > > 
> > > > > > > 意思是MAIN-REPO变量只是设置xmake-repo的包索引仓？而包的实际下载url还是根据包本身的配置信息？
> > > > > > 
> > > > > > 
> > > > > > 那当然了，xmake 不提供云存储，不可能所有包都去 github 上存一份，包自身源码在哪 依赖包本身的官方源，有可能他们放在 github，也有可能放在 sourceforge gitlab ftp 等各种地方。。怎么可能统一切
> > > > > 
> > > > > 
> > > > > 我刚刚清除了.xmake文件夹重新xmake，显示这个：
> > > > > ```shell
> > > > > checking for platform ... linux
> > > > > checking for architecture ... x86_64
> > > > > updating repositories .. error: Cloning into '/home/wq/.xmake/repositories/build-artifacts'...
> > > > > fatal: unable to access 'https://github.com/xmake-mirror/build-artifacts.git/': Could not resolve host: github.com
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > 这个是不是意味着MAIN-REPO没起作用？
> > > > 
> > > > 
> > > > 仔细看文档，二进制包仓库，不是这个 env 配置，https://xmake.io/#/zh-cn/guide/configuration?id=xmake_binary_repo
> > > 
> > > 
> > > 二进制包仓库和主仓的env是同一个，同一个变量不能同时两个值吧
> > 
> > 
> > 不是同一个 env ，XMAKE_MAIN_REPO and XMAKE_BINARY_REPO
> 
> 那就是你文档的代码写错了

刚修了

**Comment by MichaelWU0726** [comment]
> 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> 
> 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> 
> 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> 
> 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86

配置了proxy也没用
```bash
checking for platform ... linux
checking for architecture ... x86_64
updating repositories .. error: Cloning into '/home/wq/.xmake/repositories/build-artifacts'...
fatal: unable to access 'https://hub.fastgit.xyz/xmake-mirror/build-artifacts.git/': Could not resolve host: hub.fastgit.xyz
```

**Comment by waruqi** [comment]
> > 看你的log ，xmake-repo 仓库本身已经被拉下来了，都开始装 openssl 包了，说明你的配置已经生效，没啥问题
> > 只是装 openssl 包，官方源也是 github https://github.com/openssl/openssl/archive/OpenSSL_1_1_1k.zip
> > 这里下载失败而已。。仓库里面每个包的 url 源，依赖包自身，有的在 github，有的在 其他地方，各不相同，这个是切不了的
> > 只能配置 github mirror，https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86
> 
> 配置了proxy也没用
> 
> ```shell
> checking for platform ... linux
> checking for architecture ... x86_64
> updating repositories .. error: Cloning into '/home/wq/.xmake/repositories/build-artifacts'...
> fatal: unable to access 'https://hub.fastgit.xyz/xmake-mirror/build-artifacts.git/': Could not resolve host: hub.fastgit.xyz
> ```

github_mirror.lua 配置只是提供一些常用 github 服务配置，但具体它是否会失效，在哪些地区不可用，不是我这能控制的，如果用不了，自己 google 下其他的 github 镜像服务地址，自己参考 mirror 设置文档，自己改成对应的 地址。。

我们本身不提供镜像服务，hub.fastgit.xyz 这些也都是第三方的，是否可用，你得找他们 。我这不可能保证 100% 可用，之前还有两个 镜像地址，最近也失效了，hub.fastgit.xyz 这个也是后来其他人帮忙 pr 更新进来的。。

或者参考 proxy 文档，走代理配置，而不是 mirror 配置
